ex21/main.go

```diff
@@ -1,5 +1,29 @@
 package main
 
-  func main() {
+import "fmt"
 
-  }
+type Adapter interface {
+	Request() string
+}
+
+type Adaptee struct {
+	name string
+}
+
+func (a *Adaptee) SpecificRequest() string {
+	return a.name
+}
+
+type AdapterImpl struct {
```

Comment:
Go devs don't appreciate (it's community bias) additional suffixes or prefixes for type indication like `IController` or `ControllerInterfacer` or `ControllerImpl`.
If you want to define interface and implementation it should be done like `type Controller interface{` and if we don't want to expose implementation we define it with the same name `type controller struct{`. Otherwise, if we will have more than will satisfy this interface, it's better to create a separate package for those implementations.
Please refer to this https://golangbyexample.com/adapter-design-pattern-go/